lib/web_ui/lib/src/engine/text/font_collection.dart

```diff
@@ -187,19 +192,37 @@ class FontManager {
     String asset,
     Map<String, String> descriptors,
   ) {
+    Future<DomFontFace?> fontFaceLoad(DomFontFace fontFace) async {
+      try {
+        final DomFontFace loadedFontFace = await fontFace.load();
+        return loadedFontFace;
+      } catch (e) {
+        printWarning('Error while trying to load font family "$family":\n$e');
+        return null;
+      }
+    }
     // try/catch because `new FontFace` can crash with an improper font family.
     try {
       final DomFontFace fontFace = createDomFontFace(family, asset, descriptors);
-      _fontLoadingFutures.add(fontFace.load().then((_) {
-        domDocument.fonts!.add(fontFace);
-      }, onError: (dynamic e) {
-        printWarning('Error while trying to load font family "$family":\n$e');
-      }));
+      _fontLoadingFutures.add(fontFaceLoad(fontFace));
     } catch (e) {
       printWarning('Error while loading font family "$family":\n$e');
     }
   }
 
+  void registerDownloadedFonts() {
```

Comment:
I think you will have to make `registerDownloadedFonts` in the PolyfillFontManager override this with a no-op.

The reason we use `PolyfillFontManager` is because in some browsers `document.fonts` is not supported (although I just checked and it seems like font support is available in all browsers now, even mobile browsers, so maybe we can just remove `PolyfillFontManager` altogether).